### PR TITLE
Refactor chain config

### PR DESCRIPTION
**Pull request by oskarvu**
Closes FRO-1058

### CODE REVIEW

File: packages/app/src/config/chain/index.ts [hunk 1]
```diff
@@ -249,75 +231,87 @@ const chainConfig: ChainConfig = {
...
+      savingsUsdsInfoQuery: undefined,
+      chartsSupported: false,
+      inputTokens: [TokenSymbol('XDAI')],
+    },
+    farms: undefined,
   },
 }
 
 export function getChainConfigEntry(chainId: number): ChainConfigEntry {
   if (typeof import.meta.env.VITE_DEV_BASE_DEVNET_RPC_URL === 'string' && chainId === base.id) {
```
Comment (yivlad) [inline packages/app/src/config/chain/index.ts:301]: I think this if can be moved to `chainConfig` object, like
```ts
...(typeof import.meta.env.VITE_DEV_BASE_DEVNET_RPC_URL === 'string' ? [{
  id: base.id as SupportedChainId,
  ...
}] : [])
```

File: packages/app/src/config/chain/index.ts [hunk 2]
```diff
@@ -360,10 +360,11 @@ export function getChainConfigEntry(chainId: number): ChainConfigEntry {
     return {
       ...chainConfig[originChainId],
       meta: getSandboxChainMeta(chainConfig[originChainId].meta, sandboxConfig),
+      supportedPages: getSupportedPages(chainConfig[originChainId]),
```
Comment (yivlad) [inline packages/app/src/config/chain/index.ts:363]: I think `supportedPages` is used only in `usePageChainId` hook, so maybe move this logic to the hook?

File: packages/app/src/domain/savings/useSavingsTokens.ts [hunk 3]
```diff
@@ -10,18 +10,21 @@ export interface UseSavingsTokensParams {
...
 
-  const inputTokens = tokensInfo.filter(({ token }) => chainConfig.savingsInputTokens.includes(token.symbol))
-  const sDaiWithBalance = tokensInfo.findOneTokenWithBalanceBySymbol(chainConfig.sDaiSymbol)
-  const sUSDSWithBalance =
-    chainConfig.sUSDSSymbol && tokensInfo.findOneTokenWithBalanceBySymbol(chainConfig.sUSDSSymbol)
+  const inputTokens = tokensInfo.filter(({ token }) => Boolean(chainConfig.savings?.inputTokens.includes(token.symbol)))
+  const sDaiWithBalance = chainConfig.sdaiSymbol
+    ? tokensInfo.findOneTokenWithBalanceBySymbol(chainConfig.sdaiSymbol)
+    : undefined
+  const sUSDSWithBalance = chainConfig.susdsSymbol
```
Comment (EmilMalanczak) [inline packages/app/src/domain/savings/useSavingsTokens.ts:25]: susdsWithBalance

File: packages/app/src/features/actions/flavours/deposit-to-savings/logic/depositToSavingsAction.ts [hunk 4]
```diff
@@ -39,6 +39,21 @@ export function createDepositToSavingsActionConfig(
       const { token, savingsToken } = action
       const assetsAmount = toBigInt(token.toBaseUnit(action.value))
 
+      function getUsdcDepositConfig(psmActionsAddress: CheckedAddress): GetWriteConfigResult {
```
Comment (yivlad) [inline packages/app/src/features/actions/flavours/deposit-to-savings/logic/depositToSavingsAction.ts:42]: I'd extract the function as a helper and place it out of `createDepositToSavingsActionConfig` - only one path uses it, so I think it's cleaner to not include it before the switch

**Review by yivlad — APPROVED** [review]

**Review by EmilMalanczak — APPROVED** [review]

**Pull request merged**